pkg/console/starter/starter_test.go

```
@@ -95,3 +99,87 @@ func TestDeduplicateObjectReferences(t *testing.T) {
+
+	t.Run("detects when ingress becomes enabled", func(t *testing.T) {
+		configClient := fakeconfigclient.NewClientset(infraConfig, clusterVersionIngressDisabled)
+
+		detected := make(chan struct{})
+		go func() {
+			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+			defer cancel()
+			wait.PollUntilContextCancel(ctx, 100*time.Millisecond, false, func(ctx context.Context) (bool, error) {
+				infraCfg, err := configClient.ConfigV1().Infrastructures().Get(ctx, api.ConfigResourceName, metav1.GetOptions{})
+				if err != nil {
+					return false, nil
```

golangci-lint (2.12.2)

[error] error is not nil (line 156) but it returns nil

(nilerr)